pkg/gofr/container/mock_container.go

```diff
@@ -75,7 +114,114 @@ func NewMockContainer(t *testing.T) (*Container, Mocks) {
 	return container, mocks
 }
 
-type MockPubSub struct {
+func (m sqlMockDB) Select(_ context.Context, _ interface{}, query string, args ...interface{}) {
+	if m.queryWithArgs == nil || len(m.queryWithArgs) == 0 {
+		m.logger.Fatalf("Did not expect any calls for Select with query: %s", query)
+	}
+
+	lastIndex := len(m.queryWithArgs) - 1
+	expectedText := m.queryWithArgs[lastIndex].queryText
+	expectedArgs := m.queryWithArgs[lastIndex].arguments
+
+	if expectedText != query {
+		m.logger.Fatalf("expected query: %s, actual query: %s", query, expectedText)
+	}
+
+	if len(args) != len(expectedArgs) {
+		m.logger.Fatalf("expected %d args, actual %d", len(expectedArgs), len(args))
+	}
+
+	for i := range args {
+		if args[i] != expectedArgs[i] {
+			m.logger.Fatalf("expected arg %d, actual arg %d", args[i], expectedArgs[i])
```

Comment:
maybe error would be enough here

```suggestion
			m.logger.Errorf("expected arg %d, actual arg %d", args[i], expectedArgs[i])
```

This way the error message will be clearer if something fails

Reply:
Here, I would like to stop the execution instead of just an error log, as proceeding even after the query and arguments of the function do not match, would not be the ideal behaviour in case of sqlmocks.

Reply:
OK, indeed